### PR TITLE
refactor(test): add unit tests for agent/internal/crypt

### PR DESCRIPTION
**Pull request by joremysh**
* add unit test for agent/internal/crypt
* fixed typo in readme

related to #241

### CODE REVIEW

File: agent/internal/crypt/pgp_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+dL+yz5t9m1C4Ns+xKtun2BEZ2yv8KoEBY0ccLEDKMPyySyeW1z3uB7J/klW2
+D8lPOwOlj3yhj8dZZ8ItDCOZMYOlYwS/Jdho8ZAwza5poE4GYfRYQhYR6wyS
+OkAba6Xu8xCJwZ0I+Z8321OP6Ra07NfOnmeQeEcHL66+8EEHt+Wne8sMDNL/
+m9i3nF7Bw3jnCioAbVJMpKecLMKUdCg0Cw5OnG7+1WSC+lwb2OEgY5V6hHan
+=Zkso
+-----END PGP PUBLIC KEY BLOCK-----
+`
+)
+
+//const passphrase = []byte(`the passphrase of the private key`) // Passphrase of the privKey
```
Comment (nandor-magyar) [inline agent/internal/crypt/pgp_test.go:145]: deadcode?

**Review by nandor-magyar — CHANGES_REQUESTED** [review]
Please make sure that lint is executed, there are lint errors.

File: agent/internal/crypt/pgp_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+func TestDecryptSecrets(t *testing.T) {
```
Comment (robot9706) [inline agent/internal/crypt/pgp_test.go:147]: In my opinion it would be better to separate the cases into multiple tests.

**Review by robot9706 — CHANGES_REQUESTED** [review]

File: agent/internal/crypt/pgp_test.go (hunk 1 above)
Comment (joremysh) [reply agent/internal/crypt/pgp_test.go]: removed

File: agent/internal/crypt/pgp_test.go (hunk 2 above)
Comment (joremysh) [reply agent/internal/crypt/pgp_test.go:150]: split the good case and fail case to 2 functions

**Comment by polaroi8d** [comment]
@joremysh could you please sign your commits?

**Comment by joremysh** [comment]
> @joremysh could you please sign your commits?

finally, it woks now

File: agent/internal/crypt/pgp_test.go (hunk 2 above)
Comment (polaroi8d) [reply agent/internal/crypt/pgp_test.go:150]: @joremysh I think you have to split every scenario into different test cases. Collecting good test cases in 1 function as you mentioned is not the best solution.

**Review by nandor-magyar — APPROVED** [review]

**Review by polaroi8d — CHANGES_REQUESTED** [review]
Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻

**Comment by joremysh** [comment]
> Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻

@polaroi8d I think there is only 1 test case in each function now.
Which lines should I split?

**Review by robot9706 — APPROVED** [review]
LGTM 🚀

File: agent/internal/crypt/pgp_test.go (hunk 2 above)
Comment (robot9706) [reply agent/internal/crypt/pgp_test.go:150]: There's only one case in each test so it looks good to me 👍

File: agent/internal/crypt/pgp_test.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+i6i094W84JtUYkFU8YnY6KfM9+8jjkD+U20VIfHzPiknKGMU4W4a7TRBKYpU
+e46/gacPY12Du9ygLDUEi+1FZmf0/fFifv6LsKP7PNnSMIxdSFG6kzfp42kt
+GtHlLddKexkr3xuZpxfs2ToCzxptHFklXuOQDLkqBAH7a8C62Uj/grtrh83t
+MFkuY5T0DDLlM3fDyia8eaYy9wYBLImZ2Zgy3it+UmUHc5OpaeMM4NYhMB5V
+08iwmeR0tVQWeMpaue0VtGLPKHPGOknElgItP7cyVWJqEr6oM/x7uNrTpl3a
+=GvKH
+-----END PGP PUBLIC KEY BLOCK-----
+`
+	privKey = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+Version: OpenPGP.js v4.10.10
```
Comment (polaroi8d) [inline agent/internal/crypt/pgp_test.go:49]: Remove the Version and Comment section.

**Comment by polaroi8d** [comment]
> > Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻
> 
> @polaroi8d I think there is only 1 test case in each function now. Which lines should I split?

It's okay, just a tiny addition. Could you please remove the `Version` and `Comment` sections from the keys, following this PR? https://github.com/dyrector-io/dyrectorio/pull/295/files

File: agent/internal/crypt/pgp_test.go (hunk 3 above)
Comment (joremysh) [reply agent/internal/crypt/pgp_test.go]: removed

**Comment by joremysh** [comment]
> > > Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻
> > 
> > 
> > @polaroi8d I think there is only 1 test case in each function now. Which lines should I split?
> 
> It's okay, just a tiny addition. Could you please remove the `Version` and `Comment` sections from the keys, following this PR? https://github.com/dyrector-io/dyrectorio/pull/295/files

Sure, removed.

**Comment by polaroi8d** [comment]
Please rebase and I'll merge. Thanks ✋🏻

**Review by polaroi8d — APPROVED** [review]

**Comment by polaroi8d** [comment]
Good job @joremysh thanks 😀 🤝

**Pull request merged**